src/main/java/com/codeit/monew/article/naver/NaverNewsCollector.java

```diff
@@ -0,0 +1,147 @@
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Mono;
+import reactor.util.retry.Retry;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class NaverNewsCollector {
+
+  private final WebClient webClient;
+  private final ArticleRepository articleRepository;
+  private final InterestRepository interestRepository;
+  private final KeywordRepository keywordRepository;
+
+  public NaverNewsResponse searchNews(String interestName, Integer display, Integer start, String sort) {
+    log.info("네이버 API 요청 : {}", interestName);
+    return webClient.get()
+        .uri(uriBuilder -> uriBuilder
+            .path("/v1/search/news.json")
+            .queryParam("query", interestName)
+            .queryParam("display", display)
+            .queryParam("start", start)
+            .queryParam("sort", sort)
+            .build())
+        .retrieve()
+        // 429면 에러로 전달 (Retry에서 잡게)
+        .onStatus(s -> s.value() == 429, resp -> resp.createException().flatMap(Mono::error))
+        .bodyToMono(NaverNewsResponse.class)
+
+        // 429만 지수백오프로 최대 3회 재시도 (2s, 4s, 8s)
+        .retryWhen(
+            Retry.backoff(3, Duration.ofSeconds(2))
+                .filter(ex -> ex instanceof WebClientResponseException w &&
+                    w.getStatusCode() == HttpStatus.TOO_MANY_REQUESTS)
+                .transientErrors(true)
+        )
+        .block();
+  }
+
+  @Transactional
+  //@Scheduled(cron = "0 0 * * * *", zone = "Asia/Seoul")// 매 시간 정각 마다
+  @Scheduled(cron = "0 * * * * *", zone = "Asia/Seoul")// 매 분 마다
+  public void fetchAndSaveHourly() {
+    log.info("네이버 기사 수집 스케줄링 수집 시작: {}", LocalDateTime.now());
+    int display = 10;
+    int start = 1;
+    String sort = "date";
+
+    List<Interest> interests = interestRepository.findAll();
+    // 모든 관심사를 가져옴
+
+    for (Interest interest : interests) {
+      Optional<Article> lastArticle = articleRepository.findTop1ByInterestOrderByArticlePublishDateDesc(interest);
+      while(true){
+        NaverNewsResponse naverNewsResponse = searchNews(interest.getName(), display, start, sort);
+        for (NaverNewsItem naverNewsItem : naverNewsResponse.items()) {
+          Article article = buildArticleFromNaverItem(naverNewsItem, interest);
+          if(duplicateArticle(article)){
```

Comment:
if문 안에서 조건검사로 사용되는 메서드는 is를 접두사로 사용하는 경우 많아 아래 처럼 작성해봐도 좋을 것 같습니다!
```suggestion
          if(isArticleDuplicated(article)){
```